### PR TITLE
tests: all pods should be in running and none in crashloopbackoff

### PR DESCRIPTION
**Pull request by cpanato**
This is blocked by: https://jira.coreos.com/browse/ALM-423

### CODE REVIEW

File: tests/smoke/cluster_test.go [hunk 1]
```diff
@@ -94,14 +94,27 @@ func allPodsRunning(t *testing.T) error {
 
 	allReady := len(pods.Items) != 0
```
Comment (enxebre) [inline tests/smoke/cluster_test.go]: Seems like a good case for a retry function? e.g. https://github.com/coreos/tectonic-torcx/blob/6a9620014cce2cf580ad9b91e6b82f202d8eb392/internal/kube.go#L198

Comment (cpanato) [reply tests/smoke/cluster_test.go]: this test already use retry: https://github.com/cpanato/tectonic-installer/blob/dcee8e0d356b53ed163981c0b2bdf4d382ab8e5c/tests/smoke/cluster_test.go#L81

Comment (squat) [reply tests/smoke/cluster_test.go]: All test points are already wrapped in retry logic :+1:

**Comment by cpanato** [comment]
@squat @enxebre can you PTAL ?

File: tests/smoke/cluster_test.go [hunk 2]
```diff
@@ -91,14 +91,27 @@ func allPodsRunning(t *testing.T) error {
...
+	// check one more time, because in case of crashloop we might get the transition
+	time.Sleep(5 * time.Second)
+	pods, err = c.Core().Pods("").List(meta_v1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("could not list pods: %v", err)
+	}
+
+	for _, p := range pods.Items {
+		if p.Status.Phase != v1.PodRunning || p.Status.ContainerStatuses[0].State.Running == nil {
+			return errors.New("pods are not all ready")
```
Comment (squat) [inline tests/smoke/cluster_test.go]: duplicating all this code is a little dirty. can we make this whole section identical to the one above, extrapolate it into a function, and just call the function twice with a timer in the middle?

Comment (cpanato) [reply tests/smoke/cluster_test.go]: sure, will do that

**Comment by cpanato** [comment]
@squat  PTAL

**Pull request merged**